### ISSUE TITLE
how to use and what is your performance?

### ISSUE THREAD

**Issue by Jasperty** [kondilidisn/hotpotqa_rev#1]

**Comment by kondilidisn** [comment]
Hello Jasperty, 
we work on this as part of a project, we didn't submit it somewhere (nor conference neither a journal) but on the following link you can see our report of the project .
[Cracking_Hotpot__Retrieval_and_Comprehension_Methods_for_Explainable_Question_Answering.pdf](https://github.com/kondilidisn/hotpotqa_rev/files/2865100/Cracking_Hotpot__Retrieval_and_Comprehension_Methods_for_Explainable_Question_Answering.pdf)

Regarding the implementation, every folder contains files which you need to replace on the original hotpot repository (https://github.com/hotpotqa/hotpot), and the rest should work under their setting.

Hope this helped!

Best,
Nikos.

**Comment by Jasperty** [comment]
> Regarding the implementation, every folder contains files which you need to replace on the original hotpot repository (https://github.com/hotpotqa/hotpot), and the rest should work under their setting.

Thanks a lot! it helps me so much, i am only concern on support facts prediction, and your f1 score from report is very high, i would like to follow your work. 

But when i run the code from "only_sp"/ "full_original_until_sp",i get a low f1 score as your log.txt, as your report says, your get f1 score of 81, it is a pretty good score, could you tell me how to run to get this score? which folder is right?

Thanks a lot !

**Comment by kondilidisn** [comment]
I think changing the optimizer to ADAM did the trick, 

change the optimizer to this : (default learning rate)
optimizer = optim.Adam(filter(lambda p: p.requires_grad, model.parameters()))

and the average_loss to this :  (no class weighting)
nll_average = nn.CrossEntropyLoss(reduction='elementwise_mean', ignore_index=IGNORE_INDEX)

Let me know if you get the right results.

**Comment by Jasperty** [comment]
> Let me know if you get the right results.

my setting is:
1) Adam optimizer with learning rate 1e-3
2) average loss with origin weight of [0.04, 1]  (because it contains many non-supporting facts, if i set no weight then its f1 will be 0 and predicts all no sp )
3) batch size 32

i have trained for almost 4 hours for almost an epoch (on 256G cpu, 4*2080ti  gpu machine, 3 persons share this machine ), now i have evaluated for twice times,  and get these results:
1)f1 is 51.7, em is 4.5,  rec is 94.9, em is much lower than f1 
2) train loss is 1.6, dev loss is 44
what is the wrong? i fell so sad........

**Comment by kondilidisn** [comment]
-- 2 ) Do not use the weights ([0.04, 1 ]) while using Adam, these weights dropped our performance. 

-- 3) use patience  == 3 ( if the performance in the validation set does not improve in 3 sequential checks (that are being performed every 100 batches ( of size 32 )) then stop the training.

-- Change the original code where they decrease the learning rate (comment it out !) Just stop training with the patience == 3, as stopping critirion

**Comment by Jasperty** [comment]
> -- 2 ) Do not use the weights ([0.04, 1 ]) while using Adam, these weights dropped our performance.
 
Wow, your suggestion works a lot, i get a big improvement by removing weight setting, i follow your hint and get these scores:
1) Evaluation  by run.py: em 40, f1 76, recall 75, is it the right result of Adam?
2) Evaluation by script from hotpot: em 28, f1 74, prec 69, recall 85(under sp_threshold 0.3), i find that some id has no prediction so the score will be lower, but em seems drop a lot?

 it is still some difference between original score, could you provide your best working version or config setting? Thanks a lot!
